File: server/ruby/app/infrastructure/mappers/payment_details/currency_conversion_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class CurrencyConversionMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::CurrencyConversion.new
          dto.accepted_by_user = response.accepted_by_user
          dto.proposal = Infrastructure::Mappers::PaymentDetails::DccProposalMapper.from_sdk_response(response.proposal)

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/currency_conversion_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.currency_conversion import CurrencyConversion
from onlinepayments.sdk.domain.currency_conversion import CurrencyConversion as CurrencyConversionSdk
from app.infrastructure.mappers.payment_details.dcc_proposal_mapper import DccProposalMapper

class CurrencyConversionMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[CurrencyConversionSdk]) -> Optional[CurrencyConversion]:
        if response is None:
            return None

        return CurrencyConversion(
            accepted_by_user = response.accepted_by_user,
            proposal = DccProposalMapper.map_from_sdk_response(response.proposal)
        )